### PR TITLE
Completing P107 for finite spaces and adding T641

### PR DESCRIPTION
**Pull request by david20000813**
Essentially the final part of #902. In light of newly added properties, some of the finite spaces have unknown traits again. Specifically, six spaces have "Has a closed point" marked as unknown. Unfortunately, none of the additions make any of the currently known traits for these spaces redundant, as far as I can tell.

### CODE REVIEW

**Comment by prabau** [comment]
It seems that the following should be true, and is not currently derivable:

A T_0 space with a focal point must have a closed point.  (The focal point must be a closed point.)

That would allow to derive many of theses traits.

**Comment by david20000813** [comment]
@prabau Ah, true, that's definitely correct and immediately deduce 4 of these. I'll check if this theorem generalizes something currently known.

**Comment by david20000813** [comment]
Doesn't seem like it's connected to any known theorems. And no spaces will have their traits redundant due to this theorem, as far as I can tell. (S150 is the only one assigned P107 manually, but its P202 is deduced from P107. I've similarly checked those spaces that have P202 assigned manually.) I'll just rewrite this PR to add the suggested theorem and reduce the manually added traits to the two that cannot be deduced from this.

**Comment by prabau** [comment]
Wait.  Maybe the result is not true?
The dual: [has a generic point + T0 + ~ has an isolated point] seems not to be true.

[π-Base, Search for `generic point + T0 + ~isolated point`](https://topology.pi-base.org/spaces?q=generic+point+%2B+T0+%2B+%7Eisolated+point)

Maybe I am missing something.  The proposed result above would be true for finite spaces, but not sure now in general.

**Comment by david20000813** [comment]
@prabau No, it's true. The difference is an arbitrary intersection of open sets may not be open, but an arbitrary intersection of closed sets is always closed. If you have a focal point, just take the intersection of all nonempty closed sets. It has to be closed, contain all focal points, but can only contain one point as otherwise the space won't be $T_0$. This can only work for the dual theorem if the space is Alexandrov, in which case you can take arbitrary intersection of open sets and still be open.

File: theorems/T000641.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+uid: T000639
```
Comment (david20000813) [inline theorems/T000641.md:2]:
```suggestion
uid: T000641
```

**Comment by david20000813** [comment]
@prabau Suggestion implemented. (Though I realized only two of the proposed traits got automatically deduced, since only two in the search are finite.)

File: theorems/T000641.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+uid: T000641
+if:
+  and:
+  - P000001: true
+  - P000202: true
```
Comment (prabau) [inline theorems/T000641.md:6]:
```suggestion
  - P000202: true
  - P000001: true
```
Putting P202 first in the list of hypotheses, as it's the more important property of the two (looks better in the list of theorems for P202).

**Review by prabau — APPROVED** [review]

**Comment by prabau** [comment]
@david20000813 This PR is causing a git problem when doing `git pull` .  The files for the P107 traits are causing phantom differences when updating my repository on the command line.  Exactly the same problem as described in PR #649.

I don't know exactly how this happens, @StevenClontz probably knows, but it's happened in the past and the remedy is exactly what's described in PR #649.  Even better: whenever creating a new file, check the correct LF setting in the editor.

Let me know if that makes sense.  For now I'll create a PR to fix it and you can approve.

**Comment by david20000813** [comment]
@prabau I think I understand this to some extent. If you create a file from scratch, sometimes the file uses the wrong control character for the line endings? I'll be more mindful about this in the future.

**Comment by prabau** [comment]
Yeah.  All you need to do while editing a file with the wrong setting is to click on CRLF and change it to LF.

**Comment by StevenClontz** [comment]
https://github.com/pi-base/data/blob/main/.vscode/settings.json should be helping us somewhat, but it's always technically possible to set the wrong line endings. Unfortunately, GitHub's PR review system doesn't make weird line endings obvious unless you open the PR in GitHub.dev (and even then, you have to inspect each file). For now, my most clever solution is to fix things ad hoc as we're doing now.

**Pull request merged**